tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/LegacyPopTest.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Extensibility;
+using Microsoft.Identity.Test.Integration.net45.Infrastructure;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Identity.Test.Integration.NetFx.HeadlessTests
+{
+    [TestClass]
+    public class LegacyPopTests
+    {
+        [TestMethod]
+        public async Task LegacyPoPAsync()
```

Comment:
@jmprieur - this integration test shows how SAL will use the 2 extensibility points: 

- WithKeyId
- WithClientAssertion

Reply:
@bgavrilMS: Wouldn't it be better to just focus on the token acquisition but leave the protocol part of the extension, injecting a protocol handler?